tests/migrations/test_autodetector.py

```diff
@@ -1290,7 +1306,7 @@ def ask_not_null_addition(self, field_name, model_name):
                 raise Exception("Should not have prompted for not null addition")
 
         before = self.make_project_state([self.author_empty, self.publisher])
-        after = self.make_project_state([self.author_with_m2m, self.publisher])
+        after = self.make_project_state([self.author_with_m2m, self.publisher, self.author_publisher_m2m])
```

Comment:
The `_m2m` states (i.e. the auto-created through models) are added by the CreateModel and AddField operations for m2m fields, and by `ProjectState.from_apps()`. In these cases they just need to be added explicitly.